app/Device.php

```diff
@@ -711,4 +711,9 @@ public static function getItemTypes() {
         $types = DB::table('devices')->whereNotNull('item_type')->select('item_type', DB::raw('COUNT(*) as count'))->groupBy('item_type')->orderBy('count', 'desc')->get()->toArray();
         return $types;
     }
+
+    public function setProblemAttribute($value) {
+        // Map null values to empty strings to avoid Metabase problems.
+        $this->attributes['problem'] = $value === NULL ? '' : $value;
```

Comment:
Could be null coalesce operator no

`$this->attributes['problem'] = $value ?? '';

Reply:
Yes; I'm too used to working on old PHP versions, clearly.  Added.